### PR TITLE
Add TS externs.

### PR DESCRIPTION
**Pull request by bicknellr**
This PR adds externs for released packages in TypeScript's `.d.ts` format.

The original JSCompiler types from which these were converted conflate types that are used only within the package to describe internal objects or expand known built-in types for use by the package itself with types that are intended for the user of the package. The new types only expose the package's interface in the externs file and keep those intended for internal use in an `env.d.ts` file in each package's source folder.

The TS externs are included in the published packages next to their JSCompiler counterparts, but are not set as the `types` field because they aren't relevant to some of the packages' `main` module (i.e. Shady CSS).

This PR also generates a new `webcomponents-bundle.d.ts` file next to `webcomponents-bundle.js`, which bundles all of the individual polyfills' externs.

### CODE REVIEW

File: packages/custom-elements/externs/custom-elements.d.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+/**
+ * @externs
```
Comment (rictic) [inline packages/custom-elements/externs/custom-elements.d.ts:2]: here and elsewhere: I don't think this file should be marked as `@externs`. In the internal bazel rules, a ts_declaration() automatically produces externs

File: packages/custom-elements/ts_src/env.d.ts [hunk 2]
```diff
@@ -0,0 +1,84 @@
+/**
+ * @externs
```
Comment (rictic) [inline packages/custom-elements/ts_src/env.d.ts:2]: ditto, re `@externs`

File: packages/html-imports/externs/html-imports.d.ts [hunk 3]
```diff
@@ -0,0 +1,57 @@
+/**
+ * @externs
```
Comment (rictic) [inline packages/html-imports/externs/html-imports.d.ts:2]: `@externs`

File: packages/shadycss/externs/shadycss.d.ts [hunk 4]
```diff
@@ -0,0 +1,71 @@
+/**
+ * @externs
```
Comment (rictic) [inline packages/shadycss/externs/shadycss.d.ts:2]: `@externs`

File: packages/shadycss/src/env.d.ts [hunk 5]
```diff
@@ -0,0 +1,31 @@
+/**
+ * @externs
```
Comment (rictic) [inline packages/shadycss/src/env.d.ts:2]: `@externs`

File: packages/shadydom/externs/shadydom.d.ts [hunk 6]
```diff
@@ -0,0 +1,37 @@
+/**
+ * @externs
```
Comment (rictic) [inline packages/shadydom/externs/shadydom.d.ts:2]: `@externs`

File: packages/shadydom/src/env.d.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
+/**
+ * @externs
```
Comment (rictic) [inline packages/shadydom/src/env.d.ts:2]: `@externs`

File: packages/template/externs/template.d.ts [hunk 8]
```diff
@@ -0,0 +1,20 @@
+/**
+ * @externs
```
Comment (rictic) [inline packages/template/externs/template.d.ts:2]: `@externs`

File: packages/custom-elements/externs/custom-elements.d.ts [hunk 9]
```diff
@@ -0,0 +1,23 @@
...
+ * Copyright (c) 2021 The Polymer Project Authors. All rights reserved. This
+ * code may only be used under the BSD style license found at
+ * http://polymer.github.io/LICENSE.txt The complete set of authors may be found
+ * at http://polymer.github.io/AUTHORS.txt The complete set of contributors may
+ * be found at http://polymer.github.io/CONTRIBUTORS.txt Code distributed by
+ * Google as part of the polymer project is also subject to an additional IP
+ * rights grant found at http://polymer.github.io/PATENTS.txt
+ */
+
+// This file needs to be a module for the declarations to apply globally.
```
Comment (rictic) [inline packages/custom-elements/externs/custom-elements.d.ts:13]: here and elsewhere: kinda the opposite is the case. declarations in non-modules are global by default

but for consistency and our own sanity, we prefer all files to be modules

File: packages/webcomponentsjs/externs/webcomponents.d.ts [hunk 10]
```diff
@@ -0,0 +1,19 @@
+/**
+ * @externs
```
Comment (rictic) [inline packages/webcomponentsjs/externs/webcomponents.d.ts:2]: `@externs`

**Review by rictic — APPROVED** [review]
Just some documentation / clarity comments. Please do address, but otherwise LGTM

File: packages/custom-elements/externs/custom-elements.d.ts (hunk 9 above)
Comment (bicknellr) [reply packages/custom-elements/externs/custom-elements.d.ts]: Yeah, this comment doesn't make sense; updated.

File: packages/custom-elements/externs/custom-elements.d.ts (hunk 1 above)
Comment (bicknellr) [reply packages/custom-elements/externs/custom-elements.d.ts]: Removed.

File: packages/custom-elements/ts_src/env.d.ts (hunk 2 above)
Comment (bicknellr) [reply packages/custom-elements/ts_src/env.d.ts]: removed

File: packages/html-imports/externs/html-imports.d.ts (hunk 3 above)
Comment (bicknellr) [reply packages/html-imports/externs/html-imports.d.ts]: removed

File: packages/shadycss/externs/shadycss.d.ts (hunk 4 above)
Comment (bicknellr) [reply packages/shadycss/externs/shadycss.d.ts]: removed

File: packages/shadycss/src/env.d.ts (hunk 5 above)
Comment (bicknellr) [reply packages/shadycss/src/env.d.ts]: removed

File: packages/shadydom/externs/shadydom.d.ts (hunk 6 above)
Comment (bicknellr) [reply packages/shadydom/externs/shadydom.d.ts]: removed

File: packages/shadydom/src/env.d.ts (hunk 7 above)
Comment (bicknellr) [reply packages/shadydom/src/env.d.ts]: removed

File: packages/template/externs/template.d.ts (hunk 8 above)
Comment (bicknellr) [reply packages/template/externs/template.d.ts]: removed

File: packages/webcomponentsjs/externs/webcomponents.d.ts (hunk 10 above)
Comment (bicknellr) [reply packages/webcomponentsjs/externs/webcomponents.d.ts]: removed

**Pull request merged**